packages/sfui/frameworks/react/shared/utils/refs.ts

```diff
@@ -0,0 +1,13 @@
+import type * as React from 'react';
+
+export function mergeRefs<T = any>(refs: Array<React.MutableRefObject<T> | React.LegacyRef<T>>): React.RefCallback<T> {
+  return (value) => {
+    refs.forEach((ref) => {
+      if (typeof ref === 'function') {
+        ref(value);
+      } else if (ref != null) {
+        (ref as React.MutableRefObject<T | null>).current = value;
+      }
+    });
+  };
+}
```

Comment:
Looks a bit too loosely-typed. I see that `LegacyRef` might be a string - shouldn't we cover this case? Also doing `ref != null` is effectively the same as `!ref`, isn't it?

Reply:
This is literally the same [code](https://github.com/gregberge/react-merge-refs/blob/main/src/index.tsx) form the lib we've been using for a while. I don't want to change it.

And no `ref != null` returns `true` if ref is anything but null or undefined (because `null == undefined` is true), while `!ref` returns `true` for all falsy values, means for null and undefined in particular. It also doesn't return true for truthy values.
https://dorey.github.io/JavaScript-Equality-Table/